salt/utils/win_lgpo_netsh.py

```diff
@@ -66,19 +74,40 @@
                                                     store='lgpo')
 """
 
-import logging
-import os
-import re
 import socket
-import tempfile
-from textwrap import dedent
 
-import salt.modules.cmdmod
+import salt.utils.platform
+import salt.utils.win_pwsh
 from salt.exceptions import CommandExecutionError
 
-log = logging.getLogger(__name__)
-__hostname__ = socket.gethostname()
+ON_OFF = {
+    0: "OFF",
+    1: "ON",
+    2: "NotConfigured",
+    "off": "False",
+    "on": "True",
+    "notconfigured": "NotConfigured",
+}
+
+ENABLE_DISABLE = {
+    0: "Disable",
+    1: "Enable",
+    2: "NotConfigured",
+    "disable": 0,
+    "enable": 1,
+    "notconfigured": 2,
+}
+OUTBOUND = {
+    0: "NotConfigured",
+    2: "AllowOutbound",
+    4: "BlockOutbound",
+    "notconfigured": "NotConfigured",
+    "allowoutbound": "Allow",
+    "blockoutbound": "Block",
+}
+
 __virtualname__ = "netsh"
+__hostname__ = socket.gethostname()
```

Comment:
```suggestion
from functools import lru_cache

@lru_cache(maxsize=None)
def _get_hostname():
    return socket.gethostname()
```
I know this code was here before, but let's improve.
The minute anyone imports this module, you get the cost of `socket.gethostname()` why would we want to do something like that?!

Now, any reference to `__hostname__` should be replaced with `_get_hostname()`.

Reply:
Looks like we only needed the `__hostname__` for the code that was removed. So, removed this entire block.

Reply:
Even better.